src/modules/emote_menu/components/Sidebar.jsx

```diff
@@ -0,0 +1,58 @@
+import React, {useEffect, useRef} from 'react';
+import Nav from 'rsuite/lib/Nav/index.js';
+import Whisper from 'rsuite/lib/Whisper/index.js';
+import Tooltip from 'rsuite/lib/Tooltip/index.js';
+import {throttle} from 'throttle-debounce';
+import {RowHeight, WindowHeight} from '../../../constants.js';
+import styles from '../styles/sidebar.module.css';
+
+const ITEM_HEIGHT = 42;
+
+export default function Sidebar({section, onChange, providers}) {
+  const containerRef = useRef(null);
+
+  const throttleFunc = throttle(1000, () => {
+    if (section.eventKey == null) return;
+
+    const top = containerRef.current.scrollTop;
+    const index = providers.findIndex((provider) => provider.id === section.eventKey);
+    const depth = index * ITEM_HEIGHT;
+
+    let newTop;
+    if (depth < top) {
+      newTop = depth;
+    }
+    if (depth + ITEM_HEIGHT > top + WindowHeight) {
+      newTop = depth - WindowHeight + RowHeight;
+    }
+    if (newTop == null) {
+      return;
+    }
+
+    containerRef.current.scrollTo({
+      top: newTop,
+      left: 0,
+      behavior: 'smooth',
+    });
+  });
+
+  useEffect(() => throttleFunc(), [section]);
```

Comment:
are you sure this is doing what you intend?

In the react lifecycle, every time a component "rerenders" it is calling this function again.

react does some magic around storing references to stop hooks from being re-ran, but when you compute functions like `throttleFunc` in the render function it will be a completely new throttle reference every time the render function runs.

as-is, anytime `section` changes the throttle function being called is a new reference.